pkg/reconciler/build/build.go

```diff
@@ -170,6 +170,31 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 			}
 			return err
 		}
+
+		// Add a unique suffix to avoid confusion when a build
+		// is deleted and re-created with the same name.
+		// We don't use GenerateName here because k8s fakes don't support it.
+		podName, err := resources.GetUniquePodName(build.Name)
+		if err != nil {
+			return err
+		}
+
+		// update with a dummy status first to avoid race condition of another event while the pod is being created
+		build.Status = v1alpha1.BuildStatus{
+			Builder: v1alpha1.ClusterBuildProvider,
+			Cluster: &v1alpha1.ClusterSpec{
+				Namespace: build.Namespace,
+				PodName:   podName,
+			},
+			StartTime: &metav1.Time{
+				Time: time.Now(),
+			},
+		}
+		err = c.updateStatus(build)
```

Comment:
`c.updateStatus(build)` is called twice.

Reply:
doh thanks - removed the extra line